docs/developers/node-tutorial.md

```diff
@@ -0,0 +1,218 @@
+# Getting and Sending Transactions with Celestia Node
+
+In this tutorial we will cover how to use the Celestia Node API to submit and
+retrieve transactions from the Data Availability Layer by their namespace
+merkle tree.
+
+## Hardware Requirements
+
+You can find hardware requirements [here](https://docs.celestia.org/nodes/bridge-validator-node#install-golang).
+
+## Setting Up Dependencies
+
+You can follow the tutorial for setting up the dependencies [here](https://docs.celestia.org/nodes/bridge-validator-node#setting-up-your-bridge-node).
+
+## Installing Celestia App
+
+In order to begin, we would need to compile the Celestia App binary.
+
+```sh
+cd $HOME
+git clone https://github.com/celestiaorg/celestia-app.git
+cd celestia-app/
+git checkout tags/v0.4.0 -b v0.4.0
+make install
+```
+
+For more information, you can follow the steps on compiling the binary [here](https://docs.celestia.org/nodes/bridge-validator-node#install-celestia-app).
+
+Once `celestia-appd` is compiled, you can run a useful script for instantiating
+a single-node network.
+
+In the `contrib` directory, open up the file `single-node.sh` and replace its
+content with the following script:
+
+```sh
+#!/bin/sh
+
+set -o errexit -o nounset
+
+CHAINID="devnet-2"
+app="./celestia-appd"
+
+# Build genesis file incl account for passed address
+coins="1000000000000000uceles"
+$app init $CHAINID --chain-id $CHAINID 
+$app keys add validator --keyring-backend="test"
+$app keys add celes --keyring-backend="test"
+# this won't work because the some proto types are decalared twice and 
+# the logs output to stdout (dependency hell involving iavl)
+$app add-genesis-account $($app keys show validator -a --keyring-backend="test") $coins
+$app add-genesis-account $($app keys show celes -a --keyring-backend="test") $coins
+$app gentx validator 5000000000uceles --keyring-backend="test" --chain-id $CHAINID
+$app collect-gentxs
+
+# Set proper defaults and change ports
+sed 's#"tcp://127.0.0.1:26657"#"tcp://0.0.0.0:26657"#g' ~/.celestia-app/config/config.toml
+sed 's/timeout_commit = "5s"/timeout_commit = "1s"/g' ~/.celestia-app/config/config.toml
+sed 's/timeout_propose = "3s"/timeout_propose = "1s"/g' ~/.celestia-app/config/config.toml
+sed 's/index_all_keys = false/index_all_keys = true/g' ~/.celestia-app/config/config.toml
```

Comment:
assuming we are running latest and greatest, we should include the sed changing from default "full" to "validator" 
more info here: https://github.com/celestiaorg/celestia-app/issues/422